src/zope/error/tests.py

```diff
@@ -170,21 +189,93 @@ class PrincipalStub(object):
 
         err_id = getErrLog[0]['id']
         self.assertEqual(tb_text,
-                          errUtility.getLogEntryById(err_id)['tb_text'])
+                         errUtility.getLogEntryById(err_id)['tb_text'])
 
         username = getErrLog[0]['username']
         self.assertEqual(username, r"unauthenticated, \xe1, \xe1, \xe1")
 
-    def setUp(self):
-        super(ErrorReportingUtilityTests, self).setUp()
-        self.log_buffer = StringIO()
-        self.log_handler = logging.StreamHandler(self.log_buffer)
-        logging.getLogger().addHandler(self.log_handler)
+    def test_getLogEntryById_not_found(self):
+        errUtility = self.makeOne()
+        self.assertIsNone(errUtility.getLogEntryById('no such id'))
 
-    def tearDown(self):
-        logging.getLogger().removeHandler(self.log_handler)
-        super(ErrorReportingUtilityTests, self).tearDown()
+    def test_getLogin_error(self):
+        class PrincipalStub(object):
+            id = 'id'
+            title = 'title'
+            description = 'description'
+            def getLogin(self):
+                raise Exception()
+        request = TestRequest()
+        request.setPrincipal(PrincipalStub())
 
+        errUtility = self.makeOne()
+        exc_info = getAnErrorInfo("Error")
+        errUtility.raising(exc_info, request=request)
+        getErrLog = errUtility.getLogEntries()
+        self.assertEqual(1, len(getErrLog))
+
+        username = getErrLog[0]['username']
+        self.assertEqual(username,
+                         u'&lt;error getting login&gt;, id, title, description')
+
+    def test_request_items(self):
+        request = TestRequest()
+        request.items().append(('request-key', 'request-value'))
+
+        errUtility = self.makeOne()
+        exc_info = getAnErrorInfo("Error")
+        errUtility.raising(exc_info, request=request)
+        getErrLog = errUtility.getLogEntries()
+        self.assertEqual(1, len(getErrLog))
+
+        req_html = getErrLog[0]['req_html']
+        self.assertEqual(req_html, u'request-key: request-value<br />\n')
```

Comment:
Strongly tempted to add some `<`/`>`/`&` characters into `'request-value'` to make sure they're escaped properly.